cfme/services/requests.py

```diff
@@ -1,288 +1,498 @@
 # -*- coding: utf-8 -*-
-from contextlib import contextmanager
-from functools import partial
-from navmazing import NavigateToAttribute
-
+from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic.widget import Text, Table, Checkbox
+from widgetastic_manageiq import BreadCrumb, SummaryFormItem, PaginationPane
+from widgetastic_patternfly import View, Input, Tab, BootstrapTreeview, FlashMessages
+from widgetastic_manageiq import Button
+from cfme.base.login import BaseLoggedInPage
+from cfme.common.vm_views import ProvisionView, BasicProvisionFormView
 from cfme.exceptions import RequestException
-from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import (
-    Input, Region, Table, fill, flash, toolbar, match_location)
 from utils.log import logger
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+from utils.varmeth import variable
+from utils.wait import wait_for
 
-REQUEST_FINISHED_STATES = {'Migrated', 'Finished'}
-
-buttons = Region(
-    locators=dict(
-        approve="//*[@title='Approve this Request']",
-        deny="//*[@title='Deny this Request']",
-        copy="//*[@title='Copy original Request']",
-        edit="//*[@title='Edit the original Request']",
-        delete="//*[@title='Delete this Request']",
-        submit="//span[@id='buttons_on']/a[@title='Submit']",
-        cancel="//a[@title='Cancel']",
-    )
-)
-
-request_table = Table('//*[@id="list_grid"]/table')
-fields = Region(
-    locators=dict(
-        reason=Input("reason"),
-        request_list=request_table
-    )
-)
 
-match_page = partial(match_location, controller='miq_request', title='Requests')
+class Request(Navigatable):
+    """
+    Class describes request row from Services - Requests page
+    """
 
+    REQUEST_FINISHED_STATES = {'Migrated', 'Finished'}
 
-# TODO Refactor these module methods and their callers for a proper request class
-class Request(Navigatable):
-    def __init__(self, appliance=None):
+    def __init__(self, description=None, cells=None, partial_check=False, appliance=None):
+        """
+        Args:
+            description: by default we'll be checking Description column to find required row
+            cells: cells used to find required row in table
+            partial_check: greedy search or not?
+        """
         Navigatable.__init__(self, appliance=appliance)
+        self.description = description
+        self.partial_check = partial_check
+        self.cells = {'Description': self.description} if cells is None else cells
+        self.rest = self._get_request_from_rest(self.partial_check)
```

Comment:
not sure about it. request can't be created itself. it's more like result from provision/service/etc request